jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -380,6 +431,91 @@ static void initJFR() {
 				Constructor<?> constructor = dcmdStartClass.getDeclaredConstructor();
 				constructor.setAccessible(true);
 				Object dcmdStartInstance = constructor.newInstance();
+
+				/*[IF JAVA_SPEC_VERSION == 11]*/
+				String fileName = null;
+				String settings = null;
+				Long delay = null;
+				Long duration = null;
+				Boolean disk = null;
+				Long maxAge = null;
+				Long maxSize = null;
+				Boolean dumpOnExit = null;
+				String[] jfrCMDLineOptionPairs = jfrCMDLineOption.split(",");
+				for (String pair : jfrCMDLineOptionPairs) {
+					String[] configKeyValue = pair.split("=");
+					if (configKeyValue.length == 2) {
+						String key = configKeyValue[0];
+						String value = configKeyValue[1];
+						switch (key) {
+							case "filename":
+								fileName = value;
+								break;
+							case "settings":
+								settings = value;
+								break;
+							case "delay":
+								delay = convertToNanoSeconds(value);
+								break;
+							case "duration":
+								duration = convertToNanoSeconds(value);
+								break;
+							case "maxage":
+								maxAge = convertToNanoSeconds(value);
+								break;
+							case "maxsize":
+								maxSize = convertToBytes(value);
+								break;
+							case "dumponexit":
+								dumpOnExit = Boolean.valueOf(value);
+								break;
+							case "disk":
+								disk = Boolean.valueOf(value);
+								break;
+						}
+					}
+				}
+				if (null == settings) {
+					settings = "default";
+				}
+				// Convert the string to a String array
+				String[] settingsArray = new String[] { settings };
+
+				Method executeMethod = dcmdStartClass.getDeclaredMethod(
+					"execute",
+					String.class, //name
+					String[].class, //settings
+					Long.class, //delay
+					Long.class, //duration
+					Boolean.class, //disk
+					String.class, //path
+					Long.class, //maxAge
+					Long.class, //maxSize
+					Boolean.class, //dumpOnExit
+					Boolean.class //pathToGcRoots
+				);
+
+				executeMethod.setAccessible(true);
+				String results = (String) executeMethod.invoke(
+					dcmdStartInstance,
+					null,
+					settingsArray,
+					delay,
+					duration,
+					disk,
+					fileName,
+					maxAge,
+					maxSize,
+					dumpOnExit,
+					null
+				);
+				if (null != results) {
+					if (enableJFRDebug) {
+						System.out.println(results);
```

Comment:
instead of this we can just use the internal JFR logging. You can initialize it with something like

```
	private static final void setLogTag(int level) {
		try {

			Class<?> logTagClass = Class.forName("jdk.jfr.internal.LogTag");
			logTagClass.getModule().implAddExports("jdk.jfr.internal", System.class.getModule());
			Field tagSetLevel = logTagClass.getDeclaredField("tagSetLevel");
			tagSetLevel.setAccessible(true);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_SYSTEM").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_SYSTEM_EVENT").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_SYSTEM_SETTING").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_SYSTEM_BYTECODE").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_SYSTEM_PARSER").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_SYSTEM_METADATA").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_SYSTEM_STREAMING").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_SYSTEM_THROTTLE").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_METADATA").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_EVENT").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_SETTING").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_DCMD").get(logTagClass), level);
			tagSetLevel.set(logTagClass.getDeclaredField("JFR_START").get(logTagClass), level);
		} catch (Exception e) {
			throw new InternalError(e);
		}
	}
```

Im not sure what the default levels are

Reply:
Noted.  Yes, I believe above should be implemented in the JFR JNI API subscribeLogLevel.